Automattic/jetpack
# My Jetpack: Filter available licenses on #/add-license page

## Description
Currently, when visiting the `#/add-license` page in My Jetpack we are provided a list of all historical licenses for the current user (attached, unattached, and revoked). Ideally, the list should only provide valid licenses that are unattached and unrevoked. This PR filters the array of licenses so that only valid options are available to choose from.

## Proposed changes:

* Filter the `availableLicenses` array in the `AddLicenseScreen` component so we are only accessing unattached and unrevoked licenses.

### Other information:

## Jetpack product discussion


- 1201069996155224-as-1204681745241083

## Does this pull request change what data or activity we track or use?


- No

## Testing instructions:





Recreating the issue:
* Start up your Jurassic Tube on `trunk`
* Activate Protect, and proceed to `/wp-admin/admin.php?page=my-jetpack#/add-license`
* Review the list of available and verify that all licenses associated with your current user are found
   * You can validate this using the Dev API Console making a `GET` request to `wpcom/v2/jetpack-licensing/user/licenses`

Verifying the fix:
* Deactivate Jetpack
* Checkout this branch
* Install and build `packages/my-jetpack`
* Activate Jetpack once again and again proceed to the `#/add-licenses` page
* Verify that a list of only unattached and unrevoked licenses are provided

Screenshots:
Before:
<img width="1090" alt="Screen Shot 2023-05-30 at 12 59 39" src="https://github.com/Automattic/jetpack/assets/43220201/fc5fd691-3145-465c-81ca-13de9e90f813">

After:
<img width="1081" alt="Screen Shot 2023-05-30 at 13 08 43" src="https://github.com/Automattic/jetpack/assets/43220201/11f815c8-ab29-4eff-9379-facdb9970d13">
